### ISSUE TITLE
Explain in more detail what the hash function does in GlobalScope

### ISSUE THREAD

**Issue by HappyHeads** [godotengine/godot-docs#10383]
Godot 4.3

Currently the documentation only says that it "Returns the integer hash of the passed variable", someone who is new to coding might not understand what it means or where it should or should not be used. I think it should also state what algorithm it uses as it is not stated anywhere.

https://docs.godotengine.org/en/stable/classes/class_@globalscope.html#class-globalscope-method-hash

**Comment by AThousandShips** [comment]
The internals of this method is an implementation detail and I don't think it should be made explicit 

If people want specific hashing functions they should use the dedicated hashing classes instead I'd say

**Comment by tetrapod00** [comment]
I agree that the hash algorithm probably shouldn't be documented if it's not meant to be relied upon.

There might be a discrepancy here to resolve, though.
[Object.get_instance_id()](https://docs.godotengine.org/en/stable/classes/class_object.html#class-object-method-get-instance-id) does say this:
> Note: This ID is only useful during the current session. It won't correspond to a similar object if the ID is sent over a network, or loaded from a file at a later time.

[@GlobalScope.instance_from_id](https://docs.godotengine.org/en/stable/classes/class_%40globalscope.html#class-globalscope-method-instance-from-id) does not have the note.
[EncodedObjectAsID](https://docs.godotengine.org/en/stable/classes/class_encodedobjectasid.html#class-encodedobjectasid) likewise does not have the note, but does say:
> This class is used internally by the editor inspector and script debugger, but can also be used in plugins to pass and display objects as their IDs.

So there is potentially an improvement here by unifying the information from these multiple places, somehow. But we probably don't want to just copy the note verbatim, since that will increase maintenance later. I'm not familiar enough with the subject to do so, but leaving this here for anyone else who is.
